dbt/adapters/redshift/connections.py

```diff
@@ -107,14 +135,115 @@ def get_connect_method(self):
             def connect():
                 logger.debug("Connecting to redshift with username/password based auth...")
                 c = redshift_connector.connect(
-                    user=self.credentials.user, password=self.credentials.password, **kwargs
+                    user=self.credentials.user,
+                    password=self.credentials.password,
+                    **kwargs,
+                )
+                if self.credentials.role:
+                    c.cursor().execute("set role {}".format(self.credentials.role))
+                return c
+
+            return connect
+
+        elif method == RedshiftConnectionMethod.AUTH_PROFILE:
+            if not self.credentials.auth_profile:
+                raise dbt.exceptions.FailedToConnectError(
+                    "Failed to use auth profile method. 'auth_profile' must be provided."
+                )
+            if not self.credentials.region:
+                raise dbt.exceptions.FailedToConnectError(
+                    "Failed to use auth profile method. 'region' must be provided."
+                )
+
+            def connect():
+                logger.debug("Connecting to redshift with authentication profile...")
+                c = redshift_connector.connect(
+                    iam=True,
+                    access_key_id=os.environ["AWS_ACCESS_KEY_ID"],
+                    secret_access_key=os.environ["AWS_SECRET_ACCESS_KEY"],
+                    session_token=os.environ["AWS_SESSION_TOKEN"],
+                    db_user=self.credentials.user,
+                    auth_profile=self.credentials.auth_profile,
+                    **kwargs,
                 )
                 if self.credentials.role:
                     c.cursor().execute("set role {}".format(self.credentials.role))
                 return c
 
             return connect
+        elif method == RedshiftConnectionMethod.IDP:
```

Comment:
have we done any local testing of these new connection methods? Ideally we would be able to cover these with integration tests but that may be unrealistic

Reply:
Yes, I've done local testing for these new connection methods. What I've done for local testing - Got some credentials from someone who manages redshift_connector and edited the profiles.yml to connect to Redshift.

Reply:
Not sure how feasible integration tests are - since we got these credentials from another team and we don't own these credentials. Ideally we should spend some time to create these credentials ourselves so that the integration tests don't depend on another team - but that may be out of scope for right now.